### PR TITLE
Mock CAF::Path

### PR DESCRIPTION
**Pull request by stdweird**
Integration of CAF::Path with Test::Quattor, e.g. creating a mocked file using mocked FileWriter will result in a mocked basedir.
Based on #77, requires https://github.com/quattor/CAF/pull/142

### CODE REVIEW

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 1]
```diff
@@ -501,7 +594,15 @@ sub set_file_contents
...
     my ($filename, $contents) = @_;
 
-    $desired_file_contents{$filename} = "$contents";
+    if (is_directory($filename)) {
+        diag("ERROR: Cannot set_file_ontents: $filename is a directory");
+    } elsif(make_directory(dirname($filename))) {
+        $desired_file_contents{$filename} = "$contents";
+        return $desired_file_contents{$filename};
+    } else {
+        diag("ERROR: Cannot set_file_ontents: cannot create directory for $filename");
```
Comment (ned21) [inline build-scripts/src/main/perl/Test/Quattor.pm]: Typo in error message here and line 598

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 2]
```diff
@@ -725,6 +827,169 @@ sub set_caf_file_close_diff
     $caf_file_close_diff = $state ? 1 :0;
 };
 
+
+=item sane_path
+
+sanitize path
```
Comment (ned21) [inline build-scripts/src/main/perl/Test/Quattor.pm]: Define "sane" here.

**Comment by stdweird** [comment]
@ned21 remarks addressed and renamed to CAF::Path

File: build-scripts/src/main/perl/Test/Quattor.pm (hunk 1 above)
Comment (jrha) [inline build-scripts/src/main/perl/Test/Quattor.pm]: Still a typo here (set_file_**c**ontents)

**Comment by stdweird** [comment]
@jrha other typo fixed

**Comment by stdweird** [comment]
test this please

**Comment by stdweird** [comment]
@ned21 @jrha tests are ok now

**Pull request merged**